File: src/main/java/io/github/takenoha/towerdefense/persistence/BattleFundsMutationResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Result of one idempotent event-funds credit or spend operation. */
public record BattleFundsMutationResult(OperationOutcome outcome, BattleFunds funds) {
    public BattleFundsMutationResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(funds, "funds");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/BattleFundsMutationResult.kt
package io.github.takenoha.towerdefense.persistence

import kotlin.jvm.JvmRecord

/** Result of one idempotent event-funds credit or spend operation. */
@JvmRecord
data class BattleFundsMutationResult(
    val outcome: OperationOutcome,
    val funds: BattleFunds,
)
